persistence/persistence-mybatis/src/main/java/org/cbioportal/persistence/mybatis/GenericAssayMyBatisRepository.java

```diff
@@ -19,17 +20,20 @@
     private static final Logger LOG = LoggerFactory.getLogger(GenericAssayMyBatisRepository.class);
 
     @Override
+    @Cacheable(cacheResolver = "staticRepositoryCacheOneResolver", condition = "@cacheEnabledConfig.getEnabled()")
     public List<GenericAssayMeta> getGenericAssayMeta(List<String> stableIds) {
 
         return genericAssayMapper.getGenericAssayMeta(stableIds);
     }
 
     @Override
+    @Cacheable(cacheResolver = "staticRepositoryCacheOneResolver", condition = "@cacheEnabledConfig.getEnabled()")
     public List<GenericAssayAdditionalProperty> getGenericAssayAdditionalproperties(List<String> stableIds) {
         return genericAssayMapper.getGenericAssayAdditionalproperties(stableIds);
     }
 
     @Override
+    @Cacheable(cacheResolver = "staticRepositoryCacheOneResolver", condition = "@cacheEnabledConfig.getEnabled()")
```

Comment:
It would seem to me the cache keys here would be the set of molecular profile ids.  I would think we want to cache at the profileId level so that 1) different combinations of ids can still leverage cache, 2) we have a finite set of cache items, allowing us to use the static cache (instead of general).     
We can achieve thisby making an inner function that fetches by individual id and cache at that level.

However, another question: if these service level methods are just pass throughs to the data access layer (mapeprs), and there is no significant processing we save by caching here, then perhaps we should just cache at the data layer.   If that is so, then also need to evaluate the granularity of the cache there, again to make sure we cache at the id level, and not the group-of-ids level.

Reply:
@alisman updated to fetch data for each profile, I don't really understand the second part, do you mean I should enable cache on mappers instead of repositories?